lib/attr_extras/attr_initialize.rb

```diff
@@ -14,20 +16,19 @@ def apply
     validate_arity = method(:validate_arity)
     set_ivar_from_hash = method(:set_ivar_from_hash)
 
+    attributes = Attributes.new(names)
+
     klass.send(:define_method, :initialize) do |*values|
       validate_arity.call(values.length, self.class)
+      plain_values, hash_values = values.partition { |value| !value.is_a?(Hash) }
```

Comment:
Yep. The `partition` method returns an empty array for `plain_values`, and calling `zip` on an empty array does nothing. I wrote a test to make sure and it passes:

```ruby
 klass = Class.new do
   vattr_initialize :my_plain_value
 end

 example = klass.new({ :hash => 123 })
 example.my_plain_value.must_be_nil
```